src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -94,12 +94,37 @@ private function executeGuardAuthenticator($uniqueGuardKey, GuardAuthenticatorIn
                 $this->logger->debug('Calling getCredentials() on guard configurator.', array('firewall_key' => $this->providerKey, 'authenticator' => get_class($guardAuthenticator)));
             }
 
+            // abort the authenticator execution if it does not support the request
+            if (method_exists($guardAuthenticator, 'supports')) {
+                if (true !== $guardAuthenticator->supports($request)) {
+                    return;
+                }
```

Comment:
That was because i did not  handled it this way originally. In fact now it would make more sense to check the interface implementation.